Literal() method will return" unknown node" with style tag

nodes, err := doc.Find(xpath) ///html/body/div/div
	if err != nil {
		panic("failed to evaluate xpath: " + err.Error())
	}
	nlist:=nodes.NodeList()
	if len(nlist)>0{
		v,err:=nodes.NodeList().Literal()
		if err!=nil{
			fmt.Println("Literal:",err)
		}else{
			*str=v
		}
	}

i get  the error "Literal: unknown node".
when use NodeList().NodeValue   or NodeList().String()  will retrun the xml not the html  content.

is this a bug?

ubuntu 15.10 go1.5.2 libxml2.9.2

git clone -b issues/7 git@github.com:lestrrat/go-libxml2.git test1
$ go test
--- FAIL: TestRegressionGH7 (0.00s)
        Error Trace:    xml_test.go:137
	Error:		Received unexpected error "unknown node"
	Messages:	Literal() should succeed
		
FAIL
exit status 1

--- a/clib/clib.go
+++ b/clib/clib.go
@@ -866,9 +866,9 @@ func XMLNodeValue(n PtrSource) (string, error) {
 
 	var s string
 	switch XMLNodeType(nptr._type) {
-	case AttributeNode, ElementNode, TextNode, CommentNode, CDataSectionNode, PiNode, EntityRefNode:
+	case AttributeNode, ElementNode, TextNode, CommentNode, PiNode, EntityRefNode:
 		s = xmlCharToString(C.xmlXPathCastNodeToString(nptr))
-	case EntityDecl:
+	case CDataSectionNode, EntityDecl:
 		if nptr.content != nil {
 			s = xmlCharToString(C.xmlStrdup(nptr.content))
 		}

--- a/dom/dom.go
+++ b/dom/dom.go
@@ -1,7 +1,7 @@
 package dom
 
 import (
-	"errors"
+	"fmt"
 	"sync"
 
 	"github.com/lestrrat/go-libxml2/clib"
@@ -10,10 +10,11 @@ import (
 )
 
 var docPool sync.Pool
+
 func init() {
 	SetupXPathCallback()
 	docPool = sync.Pool{}
-	docPool.New = func() interface {} {
+	docPool.New = func() interface{} {
 		return Document{}
 	}
 }
@@ -69,14 +70,16 @@ func wrapText(n uintptr) *Text {
 // go-libxml2 consumers that can generate a C.xmlNode pointer to
 // create libxml2.Node types, e.g. go-xmlsec.
 func WrapNode(n uintptr) (types.Node, error) {
-	switch clib.XMLGetNodeTypeRaw(n) {
+	switch typ := clib.XMLGetNodeTypeRaw(n); typ {
 	case clib.AttributeNode:
 		return wrapAttribute(n), nil
 	case clib.ElementNode:
 		return wrapElement(n), nil
 	case clib.TextNode:
 		return wrapText(n), nil
+	case clib.CDataSectionNode:
+		return wrapCDataSection(n), nil
 	default:
-		return nil, errors.New("unknown node")
+		return nil, fmt.Errorf("unknown node: %d", typ)
 	}
 }

--- a/dom/node_text.go
+++ b/dom/node_text.go
@@ -4,6 +4,10 @@ import (
 	"github.com/lestrrat/go-libxml2/clib"
 )
 
+func (n *CDataSection) Literal() (string, error) {
+	return clib.XMLNodeValue(n)
+}
+
 // Data returns the content associated with this node
 func (n *Text) Data() string {
 	return clib.XMLTextData(n)

--- a/types/types.go
+++ b/types/types.go
@@ -1,5 +1,5 @@
 /*
-Package types exist to provide with common types that are used 
+Package types exist to provide with common types that are used
 through out in go-libxml2. This package contains mainly interfaces
 to things that are implemented else. It is in its own package
 so that any package can refer to these interfaces without introducing

--- a/xml_test.go
+++ b/xml_test.go
@@ -113,11 +113,14 @@ func TestNamespacedReconciliation(t *testing.T) {
 }
 
 func TestRegressionGH7(t *testing.T) {
-	doc, err := ParseHTMLString(`<html>
+	doc, err := ParseHTMLString(`<!DOCTYPE html>
+<html>
 <body>
-	<div>
-		<div>Hello, World!</div>
-	</div>
+<div>
+<style>
+</style>
+    1234
+</div>
 </body>
 </html>`)
 
@@ -125,7 +128,7 @@ func TestRegressionGH7(t *testing.T) {
 		return
 	}
 
-	nodes := xpath.NodeList(doc.Find(`./body/div/div`))
+	nodes := xpath.NodeList(doc.Find(`./body/div`))
 	if !assert.NotEmpty(t, nodes, "Find should succeed") {
 		return
 	}

--- a/xml_test.go
+++ b/xml_test.go
@@ -8,6 +8,7 @@ import (
 	"github.com/lestrrat/go-libxml2/dom"
 	"github.com/lestrrat/go-libxml2/parser"
 	"github.com/lestrrat/go-libxml2/types"
+	"github.com/lestrrat/go-libxml2/xpath"
 	"github.com/stretchr/testify/assert"
 )
 
@@ -109,4 +110,32 @@ func TestNamespacedReconciliation(t *testing.T) {
 	*/
 
 	t.Logf("%s", d.String())
+}
+
+func TestRegressionGH7(t *testing.T) {
+	doc, err := ParseHTMLString(`<html>
+<body>
+	<div>
+		<div>Hello, World!</div>
+	</div>
+</body>
+</html>`)
+
+	if !assert.NoError(t, err, "ParseHTMLString should succeed") {
+		return
+	}
+
+	nodes := xpath.NodeList(doc.Find(`./body/div/div`))
+	if !assert.NotEmpty(t, nodes, "Find should succeed") {
+		return
+	}
+
+	v, err := nodes.Literal()
+	if !assert.NoError(t, err, "Literal() should succeed") {
+		return
+	}
+	if !assert.NotEmpty(t, v, "Literal() should return some string") {
+		return
+	}
+	t.Logf("v = '%s'", v)
 }

